### ISSUE TITLE
DataFrame drop_column should be replaced with drop and not be inplace

### ISSUE THREAD

**Issue by kkraus14** [rapidsai/cudf#370]
Usage should be along the lines of:

```
gdf = cudf.DataFrame()
gdf['a'] = [1,2,3,4,5]
gdf['b'] = [11,12,13,14,15]
gdf['c'] = [21,22,23,24,25]

gdf = gdf.drop(['b', 'c']) # assumes axis=1 and inplace=False compared to Pandas
```

**Comment by bahaugen** [comment]
I think I will try to make an attempt at this issue but I have a few followup questions.

1. Is `def drop(self, labels, inplace=False)` the desired method signature?
2. Do you want to leave the old drop_column method in place or should that be removed?

**Comment by kkraus14** [comment]
@bahaugen I believe @thomcom is already working on this, but I'll let him chime in.

1. We don't have plans to have inplace operations currently so we can leave the inplace parameter out but note in the docs that it is not an inplace operation.
2. We should add a deprecation warning to it that it will be removed in a future release for now.

**Comment by thomcom** [comment]
Hey @bahaugen I am at work on this already, sorry for not updating on that progress. I'm not finished yet, hopefully this evening. :) How's your progress?

**Comment by bahaugen** [comment]
Hi @thomcom I started working on it a bit last night before I knew you weren't supporting  inplace operations. You can find it here 
https://github.com/bahaugen/cudf/tree/dataframe_drop_method

I am completely new to the project so I am not sure I am even on the right path. Feel free to use any of it you would like.

**Comment by thomcom** [comment]
Thanks @bahaugen ! Go ahead and submit a PR and we will review it and make any (if) necessary changes. I am always loathe to ignore good work. :)

**Comment by kkraus14** [comment]
Closing this as a duplicate of #466 and merged in #439.
